ios/BurntModule.swift

```diff
@@ -107,118 +127,146 @@ struct ToastOptions: Record {
-
+  
   @Field
   var preset: ToastPreset = ToastPreset.done
-
+  
   @Field
   var duration: TimeInterval?
-
+  
   @Field
   var layout: ToastLayout?
-
+  
   @Field
   var shouldDismissByDrag: Bool = true
-
+  
   @Field
   var haptic: ToastHaptic = .none
-
+  
   @Field
   var from: ToastPresentSide = .top
+  
+  @Field
+  var icon: Icon? = nil
 }
 
-enum ToastHaptic: String, EnumArgument {
+enum ToastHaptic: String, Enumerable {
   case success
   case warning
   case error
   case none
-
+  
   func toSPIndicatorHaptic() -> SPIndicatorHaptic {
     switch self {
-    case .success:
-      return .success
-    case .warning:
-      return .warning
-    case .error:
-      return .error
-    case .none:
-      return .none
+      case .success:
+        return .success
+      case .warning:
+        return .warning
+      case .error:
+        return .error
+      case .none:
+        return .none
     }
   }
 }
 
-enum ToastPreset: String, EnumArgument {
+enum ToastPreset: String, Enumerable {
   case done
   case error
-
-  func toSPIndicatorPreset() -> SPIndicatorIconPreset {
+  case none
+  case custom
+  
+  func toSPIndicatorPreset(_ options: ToastOptions?) -> SPIndicatorIconPreset? {
     switch self {
-    case .done:
-      return .done
-    case .error:
-      return .error
+      case .done:
+        return .done
+      case .error:
+        return .error
+      case .custom:
+        return .custom(UIImage.init( systemName: options?.icon?.name ?? "swift")!.withTintColor(options?.icon?.color ?? .systemBlue, renderingMode: .alwaysOriginal))
+      case .none:
+        return .none
+        
     }
   }
 }
 
-enum ToastPresentSide: String, EnumArgument {
+enum ToastPresentSide: String, Enumerable {
   case top
   case bottom
-
+  
   func toSPIndicatorPresentSide() -> SPIndicatorPresentSide {
     switch self {
-    case .top:
-      return .top
-    case .bottom:
-      return .bottom
+      case .top:
+        return .top
+      case .bottom:
+        return .bottom
     }
   }
 }
 
 public class BurntModule: Module {
   public func definition() -> ModuleDefinition {
     Name("Burnt")
-
+    
     AsyncFunction("toastAsync") { (options: ToastOptions) -> Void in
-      let view = SPIndicatorView(title: options.title, message: options.message, preset: options.preset.toSPIndicatorPreset())
-
+      let view : SPIndicatorView
+      if(options.preset == .none){
+        view = SPIndicatorView(title: options.title, message: options.message)
+      } else if(options.preset == .custom){
+        view = SPIndicatorView(title: options.title, message: options.message, preset: options.preset.toSPIndicatorPreset(options)!)
+      }
+      else{
+        view = SPIndicatorView(title: options.title, message: options.message, preset: options.preset.toSPIndicatorPreset(nil)!)
+      }
+      
       if let duration = options.duration {
         view.duration = duration
       }
-
+      
       if let icon = options.layout?.iconSize {
```

Comment:
This line is also wrong. We're not using `options.layout` anymore. It's `icon.height` now.

It should be something like:

```swift
if let height = options.icon?.height, let width = options.icon?.width {
    view.layout.iconSize = .init(width: width, height: height)
}
```

Reply:
accurately the `icon.ios` object is only available in the "custom" preset, so if we move `height` and `width` to the `icon.ios` object will cause the other preset type to unavailable, even if we can change the type, but I feel it will be a little bit messy.